frontends/web/src/components/sidebar/sidebar.tsx

```diff
@@ -75,12 +78,26 @@ const eject = (e: React.SyntheticEvent): void => {
 
 const Sidebar = ({
   deviceIDs,
+  devices,
   accounts,
 }: SidebarProps) => {
   const { t } = useTranslation();
   const { pathname } = useLocation();
+  const [ canUpgrade, setCanUpgrade ] = useState(false);
   const { activeSidebar, sidebarStatus, toggleSidebar } = useContext(AppContext);
 
+  useEffect(() => {
+    const deviceList = Object.entries(devices).filter(([_, productName]) => productName === 'bitbox02');
+    if (deviceList.length) {
+      Promise.all(
```

Comment:
It does not matter much with only one device, but if we had multiple, you'd want to not call getVersion on all of them but abort early if one returns true.

I asked ChatGPT to rewrite it with this in mind and it spit out this:

```
useEffect(() => {
  const checkUpgradableDevices = async () => {
    const bitboxDevices = Object.keys(devices).filter(deviceID => devices[deviceID] === 'bitbox02');

    for (const deviceID of bitboxDevices) {
      const { canUpgrade } = await getVersion(deviceID);
      if (canUpgrade) {
        setCanUpgrade(true);
        return; // Exit early as we found an upgradable device
      }
    }
  };

    checkUpgradableDevices();
}, [devices]);
```

imo this is also clearer to read. Promise.all etc. are a bit hard to read for me.

wdyt?

Reply:
hm... Now we have the opposite problem, I find it hard to read. It was my intention to do the calls in parallel and set state as soon as the first result returns true.

I guess am not used to await in for loops and find it not natural (yet).  I should see how this transpiles to Promises to really understand. 

By reading your/gpt version I can't even tell if it is parallel or in sequence.

I'll try to understand your version if you read Promise.all doc :)
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

Reply:
@benma changed to async / sequential calls, tested with 1 device :)

<img width="250" alt="Screenshot 2024-06-13 at 08 27 17" src="https://github.com/BitBoxSwiss/bitbox-wallet-app/assets/546900/bf218024-b028-48cc-96de-8366c976f206">

<img width="340" alt="Screenshot 2024-06-13 at 08 27 30" src="https://github.com/BitBoxSwiss/bitbox-wallet-app/assets/546900/f8652917-e60a-4a7b-9172-97c3a784e6d6">

Reply:
I actually didn't consider concurrent vs sequential execution at all :flushed: :thinking: 

The for-loop is sequential. Imo it's fine to keep your newest version even though it is not concurrent, as it's imho much clearer, and a bit less needless load on the backend (if we ever have multiple device support :joy:).